### PR TITLE
Kh/opencv integration

### PR DESCRIPTION
**Pull request by madhephaestus**
This moves the Unique Identity into the kernel to keep the memory from exploding. 

test with https://github.com/madhephaestus/YoloJava.git

### CODE REVIEW

**Comment by JansenSmith** [comment]
Looks like I'm getting multiple errors in Eclipse before running:
image

and also runtime errors:
```
Skip Writing file contents, file is same
script error
java.lang.Exception: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 46: unable to resolve class org.opencv.core.Mat
 @ line 46, column 1.
   import org.opencv.core.Mat;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 50: unable to resolve class org.opencv.core.Size
 @ line 50, column 1.
   import org.opencv.core.Size;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 52: unable to resolve class org.opencv.objdetect.CascadeClassifier
 @ line 52, column 1.
   import org.opencv.objdetect.CascadeClassifier;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 49: unable to resolve class org.opencv.core.Scalar
 @ line 49, column 1.
   import org.opencv.core.Scalar;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 54: unable to resolve class org.opencv.videoio.VideoCapture
 @ line 54, column 1.
   import org.opencv.videoio.VideoCapture;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 44: unable to resolve class org.opencv.core.Core
 @ line 44, column 1.
   import org.opencv.core.Core;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 48: unable to resolve class org.opencv.core.Rect
 @ line 48, column 1.
   import org.opencv.core.Rect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 53: unable to resolve class org.opencv.objdetect.Objdetect
 @ line 53, column 1.
   import org.opencv.objdetect.Objdetect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 45: unable to resolve class org.opencv.core.Point
 @ line 45, column 1.
   import org.opencv.core.Point;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 47: unable to resolve class org.opencv.core.MatOfRect
 @ line 47, column 1.
   import org.opencv.core.MatOfRect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 51: unable to resolve class org.opencv.imgproc.Imgproc
 @ line 51, column 1.
   import org.opencv.imgproc.Imgproc;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 196: unable to resolve class org.opencv.core.Point 
 @ line 196, column 26.
   				HashMap<UniquePerson,org.opencv.core.Point> lhm =  upf.getCurrentPersons()
                            ^

12 errors

	at com.neuronrobotics.bowlerstudio.scripting.ScriptingFileWidget$2.run(ScriptingFileWidget.java:337)
Caused by: org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 46: unable to resolve class org.opencv.core.Mat
 @ line 46, column 1.
   import org.opencv.core.Mat;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 50: unable to resolve class org.opencv.core.Size
 @ line 50, column 1.
   import org.opencv.core.Size;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 52: unable to resolve class org.opencv.objdetect.CascadeClassifier
 @ line 52, column 1.
   import org.opencv.objdetect.CascadeClassifier;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 49: unable to resolve class org.opencv.core.Scalar
 @ line 49, column 1.
   import org.opencv.core.Scalar;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 54: unable to resolve class org.opencv.videoio.VideoCapture
 @ line 54, column 1.
   import org.opencv.videoio.VideoCapture;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 44: unable to resolve class org.opencv.core.Core
 @ line 44, column 1.
   import org.opencv.core.Core;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 48: unable to resolve class org.opencv.core.Rect
 @ line 48, column 1.
   import org.opencv.core.Rect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 53: unable to resolve class org.opencv.objdetect.Objdetect
 @ line 53, column 1.
   import org.opencv.objdetect.Objdetect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 45: unable to resolve class org.opencv.core.Point
 @ line 45, column 1.
   import org.opencv.core.Point;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 47: unable to resolve class org.opencv.core.MatOfRect
 @ line 47, column 1.
   import org.opencv.core.MatOfRect;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 51: unable to resolve class org.opencv.imgproc.Imgproc
 @ line 51, column 1.
   import org.opencv.imgproc.Imgproc;
   ^

C:\Users\lenovo\bowler-workspace\gitcache\github.com\madhephaestus\YoloJava\ResnetWithCamera.groovy: 196: unable to resolve class org.opencv.core.Point 
 @ line 196, column 26.
   				HashMap<UniquePerson,org.opencv.core.Point> lhm =  upf.getCurrentPersons()
                            ^

12 errors

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:309)
	at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:938)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:585)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:534)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:286)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:259)
	at groovy.lang.GroovyShell.parseClass(GroovyShell.java:674)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:686)
	at groovy.lang.GroovyShell.parse(GroovyShell.java:695)
	at com.neuronrobotics.bowlerstudio.scripting.GroovyHelper.inline(GroovyHelper.java:57)
	at com.neuronrobotics.bowlerstudio.scripting.GroovyHelper.inlineScriptRun(GroovyHelper.java:81)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingEngine.inlineScriptRun(ScriptingEngine.java:442)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingEngine.inlineFileScriptRun(ScriptingEngine.java:950)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingFileWidget$2.run(ScriptingFileWidget.java:319)
```

**Comment by madhephaestus** [comment]
Did you refresh gradle?

File: build.gradle [hunk 1]
```diff
@@ -336,8 +336,8 @@ dependencies {
 	
 	// https://mvnrepository.com/artifact/com.alphacephei/vosk
 	implementation group: 'com.alphacephei', name: 'vosk', version: '0.3.45'
-	
-
+	// Add OpenCV 
+	implementation 'org.openpnp:opencv:4.7.0-0'
```
Comment (madhephaestus) [inline build.gradle:340]: @JansenSmith changes to the build.gradle means you need to refresh the gradle, any time this file is changed in a PR, refresh the gradle.

Comment (JansenSmith) [reply build.gradle:340]: that was certainly an issue, thank you

**Comment by JansenSmith** [comment]
@madhephaestus currently looking into this issue:
```
Starting Yolo PyTorch Java example
groovy.lang.MissingPropertyException: No such property: BufferedImageUtils for class: YoloJava
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:51)
	at org.codehaus.groovy.runtime.callsite.PogoGetPropertySite.getProperty(PogoGetPropertySite.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGroovyObjectGetProperty(AbstractCallSite.java:231)
	at YoloJava.run(YoloJava.groovy:18)
	at com.neuronrobotics.bowlerstudio.scripting.GroovyHelper.inline(GroovyHelper.java:61)
	at com.neuronrobotics.bowlerstudio.scripting.GroovyHelper.inlineScriptRun(GroovyHelper.java:81)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingEngine.inlineScriptRun(ScriptingEngine.java:442)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingEngine.inlineFileScriptRun(ScriptingEngine.java:950)
	at com.neuronrobotics.bowlerstudio.scripting.ScriptingFileWidget$2.run(ScriptingFileWidget.java:319)
```

**Comment by JansenSmith** [comment]
@madhephaestus should it be resolving BufferedImageUtils from the import java.awt.image.BufferedImage
?

**Comment by JansenSmith** [comment]
@madhephaestus should it be resolving BufferedImageUtils from the import java.awt.image.BufferedImage
?

**Review by JansenSmith — APPROVED** [review]
tested and reviewed codebase

**Pull request merged**